src/manager/manager.py

```diff
@@ -62,10 +38,47 @@ def _update_manager_vehicle_list(manager: Manager, vehicles: list[Vehicle]):
         # append if not in list and inside radius
         if not vehicle_in_list and distance_to_vehicle <= manager.radius: 
             manager.vehicles.append(vehicle)
+            new_vehicle = True
 
         # remove if in list and outside radius
         elif vehicle_in_list and distance_to_vehicle > manager.radius: 
             manager.vehicles.remove(vehicle)
+    return new_vehicle
+
+def get_collisions(manager: Manager, cur_time: float):
+    collisions = []
+    vehicle_pairs = combinations(manager.vehicles, 2)
+    threshold_distance = 2.5 # will be replaced by Alex's defined safety radius
+    
+    for vehicle_pair in vehicle_pairs:
+        vehicle_out_of_bounds_time = int(min(time_until_end_of_route(vehicle_pair[0]), time_until_end_of_route(vehicle_pair[1])))
+        def world_pos_0(t):
+            return route_position_to_world_position(vehicle_pair[0].route, route_position_at_time(vehicle_pair[0], t))
+        def world_pos_1(t):
+            return route_position_to_world_position(vehicle_pair[1].route, route_position_at_time(vehicle_pair[1], t))
+        def distance(t):
```

Comment:
Can be a one-liner once you have the functions for it, or separate distance function